internal/pkg/template/templates/workloads/partials/cf/service-base-properties.yml

```
@@ -47,7 +47,9 @@ NetworkConfiguration:
         - Fn::ImportValue: !Sub '${AppName}-${EnvName}-{{.Network.SubnetsType}}'
     {{- end}}
     SecurityGroups:
+      {{- if not .Network.DenyDefaultSecurityGroup}}
```

Nice! Just to put my my thoughts here in case others are wondering.

The deployment will error out if there turns out to be zero security groups (i.e. the default is denied, no custom sg, no NLB and no addon sg - should be rare), because the template is malformed with `SecurityGroups: ` field having `null` value.

However, if we remove `SecurityGroups` field (so template isn't malformed and the deployment goes through), the [default security group of the VPC will be used](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/default-custom-security-groups.html), which I think we want to avoid. So I think keeping `SecurityGroups` field here and just let it error out if no sg is left is the better option!